src/dataframe/dataframe.jl

```diff
@@ -278,6 +278,28 @@ _columns(df::DataFrame) = getfield(df, :columns)
 nrow(df::DataFrame) = ncol(df) > 0 ? length(_columns(df)[1])::Int : 0
 ncol(df::DataFrame) = length(index(df))
 
+##############################################################################
+##
+## DataFrame consistency check
+##
+##############################################################################
+
+function _check_consistency(df::DataFrame)
+    cols, idx = _columns(df), index(df)
+    ncols = length(cols)
+    @assert length(idx.names) == length(idx.lookup) == ncols
+    ncols == 0 && return nothing
+    nrows = length(cols[1])
+    for i in 2:length(cols)
+        @assert length(cols[i]) == nrows "Length of column $(names(df)[i]) ($(length(df[!, i])))" *
+                                         " does not match length of column 1 ($(length(df[!, 1]))). " *
+                                         "It is likely caused by unintended resizing of the column."
```

Comment:
```suggestion
                                         "The column vector has likely been resized unintentionally " *
                                         "(either directly or because it is shared with another data frame)."
```